eslint.config.js

```diff
@@ -0,0 +1,309 @@
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: ['@grafana/runtime', '@grafana/ui', '@grafana/data'],
+        },
+      ],
+    },
+  },
+  {
+    name: 'grafana/ui-overrides',
+    files: ['packages/grafana-ui/**/*.{ts,tsx}'],
+    ignores: ['packages/grafana-ui/**/*.{test,story}.{ts,tsx}'],
+    rules: {
+      '@emotion/syntax-preference': [2, 'object'],
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: ['@grafana/runtime', '@grafana/data/*', '@grafana/ui', '@grafana/e2e-selectors/*'],
+          paths: [
+            {
+              name: 'react-i18next',
+              importNames: ['Trans', 't'],
+              message: 'Please import from grafana-ui/src/utils/i18n instead',
+            },
+          ],
+        },
+      ],
+    },
+  },
+  {
+    name: 'grafana/schema-overrides',
+    files: ['packages/grafana-schema/**/*.{ts,tsx}'],
+    ignores: ['packages/grafana-schema/**/*.test.{ts,tsx}'],
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: ['@grafana/*'],
+        },
+      ],
+    },
+  },
+  {
+    name: 'grafana/runtime-overrides',
+    files: ['packages/grafana-runtime/**/*.{ts,tsx}'],
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: ['@grafana/runtime', '@grafana/data/*', '@grafana/ui/*', '@grafana/e2e/*'],
+        },
+      ],
+    },
+  },
+  {
+    name: 'grafana/flamegraph-overrides',
+    files: ['packages/grafana-flamegraph/**/*.{ts,tsx}'],
+    ignores: ['packages/grafana-flamegraph/**/*.{test,story}.{ts,tsx}'],
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: ['@grafana/runtime', '@grafana/e2e', '@grafana/e2e-selectors/*'],
+        },
+      ],
+    },
+  },
+  {
+    name: 'grafana/alerting-overrides',
+    ...testingLibraryPlugin.configs['flat/react'],
+    files: ['public/app/features/alerting/**/*.{ts,tsx,js,jsx}'],
+    plugins: {
+      'testing-library': testingLibraryPlugin,
+      'jest-dom': require('eslint-plugin-jest-dom'),
+    },
+    rules: {
+      'dot-notation': 'error',
+      'prefer-const': 'error',
+      'react/no-unused-prop-types': 'error',
+    },
+  },
+  {
+    name: 'grafana/alerting-test-overrides',
+    ...testingLibraryPlugin.configs['flat/react'],
+    files: [
+      'public/app/features/alerting/**/__tests__/**/*.[jt]s?(x)',
+      'public/app/features/alerting/**/?(*.)+(spec|test).[jt]s?(x)',
+    ],
+    rules: {
+      'testing-library/prefer-user-event': 'error',
+      'jest/expect-expect': ['error', { assertFunctionNames: ['expect*', 'reducerTester'] }],
+    },
+  },
+  {
+    name: 'grafana/explore-traceview-overrides',
+    files: ['public/app/features/explore/TraceView/components/demo/**/*.{ts,tsx,js,jsx}'],
+    rules: {
+      'import/no-extraneous-dependencies': 'off',
+    },
+  },
```

Comment:
The rules aren't all being applied here - `testing-library` needs compatibility changes in order to work, and the rules need to be spread differently. I've made the changes locally and can add to this PR if you're happy for me to do so.

Might be worth us double checking that rules will still fail in the different places that used to have separate configs

Reply:
Please feel free to push to the PR @tomratcliffe. I'd appreciate the assistance as it's already been more time consuming than I thought to get this far. 😅